### PR TITLE
fix(Dropdown): Pass initialSelectedId to CoreDropdown when it updates…

### PR DESCRIPTION
**Pull request by GuyPerkal**
… as well as initially

I wanted to add tests but I can't figure out how to update initialSelectedId after setting it initially in tests, updating a variable didn't work. Example of test I want to write - 

```js
  it('should update selected id when initial selected id changes', async () => {
    const options = new Array(5).fill(null).map((el, i) => ({
      id: `${i}`,
      value: `value-${i}`,
      isSelectable: i < 3,
    }));
    let selectedId = '0';
    const driver = createDriver(
      <Dropdown options={options} initialSelectedId={selectedId} />,
    );
    expect(await driver.getDropdownText()).toEqual(options[0].value);
    selectedId = '1';
    expect(await driver.getDropdownText()).toEqual(options[1].value);
  });
```

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_513/

**Comment by GuyPerkal** [comment]
See also this PR - https://github.com/wix/wix-ui/pull/1977
Not updating CoreDropwdown's `initialSelectedIds` can cause its selectedIds to be out of sync with its options when they update.

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -94,7 +94,10 @@ export class Dropdown extends React.Component<DropdownProps, State> {
   }
 
   static getDerivedStateFromProps(nextProps, state) {
-    if (state.selectedOption) {
+    if (
+      state.selectedOption &&
+      state.selectedOption.id !== nextProps.initialSelectedId
+    ) {
```
Comment (jonathanadler) [inline src/components/Dropdown/Dropdown.tsx:100]: Did you mean 
```suggestion
    if (
      state.selectedOption &&
      state.selectedOption.id === nextProps.initialSelectedId
    ) {
```
?

Comment (GuyPerkal) [reply src/components/Dropdown/Dropdown.tsx]: Fixed 😳

**Review by jonathanadler — APPROVED** [review]
Thanks!

I can merge, but I think we'll only publish this in the beginning  of next week, is that ok?

**Comment by GuyPerkal** [comment]
Sure, the more important fix was the one you merged on wix-ui-core.

**Comment by GuyPerkal** [comment]
And anyway I see the tests failed so I will also look at it on Sunday :)

**Comment by GuyPerkal** [comment]
Should be OK to merge now.

**Comment by jonathanadler** [comment]
> Should be OK to merge now.

Awesome, let's wait for the builds 💪

**Pull request merged**